eventuals/static-thread-pool.h

```diff
@@ -160,48 +181,30 @@ struct _StaticThreadPoolSchedule {
 
       auto& pinned = requirements()->pinned;
 
-      if (!pinned.core) {
-        // TODO(benh): pick the least loaded core. This will require
-        // iterating through and checking the sizes of all the "queues"
-        // and then atomically incrementing which ever queue we pick
-        // since we don't want to hold a lock here.
-        pinned.core = 0;
-      }
-
-      assert(pinned.core);
-
-      if (!(*pinned.core < pool()->concurrency)) {
-        CHECK(!adapted_);
-        if (interrupt_ != nullptr) {
-          k_.Register(*interrupt_);
-        }
-        k_.Fail("'" + context_->name() + "' required core is > total cores");
+      if (StaticThreadPool::member && StaticThreadPool::cpu == pinned.CPU()) {
+        Adapt();
+        auto* previous = Scheduler::Context::Switch(context_.get());
+        adapted_->Start(std::forward<Args>(args)...);
+        previous = Scheduler::Context::Switch(previous);
+        CHECK_EQ(previous, context_.get());
```

Comment:
Let's revert this because _this_ is when we want to pick the least loaded core. NOT when we're creating a `Pinned`.